### PR TITLE
chore: update dependencies to the latest version

### PR DESCRIPTION
**Pull request by ersachin3112**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [x] **metadata update**

### Motivation / Use-Case
update dependencies to the latest version

### Breaking Changes
No

### Additional Info
No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,40 +57,40 @@
...
     "jest": "^29.7.0",
     "lint-staged": "^15.2.2",
     "npm-run-all": "^4.1.5",
     "postcss-scss": "^4.0.9",
     "prettier": "^3.2.5",
     "standard-version": "^9.5.0",
-    "stylelint": "^16.2.1",
-    "typescript": "^5.4.2",
-    "webpack": "^5.90.3"
+    "stylelint": "~16.3.1",
```
Comment (alexander-akait) [inline package.json:91]: Why it was changed?

**Review by alexander-akait — CHANGES_REQUESTED** [review]
Some questions

File: package.json (hunk 1 above)
Comment (ricardogobbosouza) [reply package.json:91]: Version 16.4.0 has typing problems

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -64,9 +64,6 @@ jobs:
         node-version: [18.x, 20.x, 21.x]
         stylelint-version: [13.x, 14.x, 15.x, 16.x]
         webpack-version: [latest]
-        exclude:
-        - stylelint-version: 16.x
-          node-version: 18.x
```
[the remark this answers is not in the document]
Comment (ricardogobbosouza) [reply .github/workflows/nodejs.yml:69]: Had problems with the action `wagoid/commitlint-github-action@v5`

Which has been fixed https://github.com/webpack-contrib/stylelint-webpack-plugin/pull/349/files

File: package.json (hunk 1 above)
Comment (ricardogobbosouza) [reply package.json:91]: https://github.com/stylelint/stylelint/blob/main/types/stylelint/index.d.ts#L132

https://github.com/cosmiconfig/cosmiconfig/blob/main/src/types.ts#L8C4-L14C10

Comment (ricardogobbosouza) [reply package.json:91]: The new version fixes the problem https://github.com/stylelint/stylelint/releases/tag/16.5.0

**Pull request merged**